### ISSUE TITLE
FieldExecution: #NN: [NN] pending... (waiting for #NN)

### ISSUE THREAD

**Issue by DarrenCook** [jpillora/verifyjs#5]
Verify.js is stopping my form being submitted, and the only clue is I keep getting that warning message, about waiting for something, in the console.log.

I've tried a few variations, but as one example I added this line to my form:
<label for="whatever">TEST:</label><input type="text" name="whatever" value="" data-validate="number">

And then, after submitting, in the log I see:
verify.js: FieldExecution: #9: [whatever] skip (not required)
verify.js: FieldExecution: #10: [3] pending... (waiting for #7)

That is the only data-validate line in the form. If I remove that line, my form goes back to submitting with no problems.

)

**Comment by bhushan-firake** [comment]
First of all, thank you for building this easy to use and awesome library. But I am facing this same issue in my code. 

I have noticed that the issue is related to the browser. It produces this error in FireFox 32.0. This issue doesn't come in Chrome.
![untitled](https://cloud.githubusercontent.com/assets/4662760/4174411/87fedf7e-358c-11e4-9e29-309bea0dcd0d.png)

**Comment by ht512512** [comment]
Has anyone found a solution or work-around? I have started to debug it but wanted to check before I get deeper.

Thanks.

**Comment by bhushan-firake** [comment]
@ht512512  I am no script guy. Wanted to use this as it is very easy to add in my code in c#. However, I have not yet resolved this error. Will update or submit a pull request if solved. Hope you will do the same.

Thank you

**Comment by ht512512** [comment]
@BhushanFirake The author has marked this issues as a bug so he is aware of it. The previous version does not have this bug so my current workaround is to fall back to the previous version. (Go to the verify.js home and "view source" and download the version included in that. )

**Comment by jpillora** [comment]
Hey guys, sorry about the lack of support, as mentioned in #7 I still keep track of my projects, though I don't have much time to fix bugs, hence I'm looking for maintainers.

To give you guys an idea of the crux of the project.
- Form objects contain form validation state and a list of element objects
- Element objects contain element validation state and a list of validators
- Validators are functions which return true or error string
- <Type>Execution objects are basically just wrappers around jQuery promises 
- FormExecution objects start FieldExecutions and GroupExecutions in parallel
- FieldExecutions objects start Validators in series

This diagram attempts to show this flow:

![validation flow](https://docs.google.com/drawings/d/1tfqBdTHzluwbNgXnAlP08WQ5wBekoCIA0aBZ4sAPC-E/pub?w=389&h=406)

It'd be great if someone wanted to volunteer to be a maintainer, and if someone is willing, I've start a contributor wiki and move this summary there and expand on it where needed.

I wrote this library in 2012, so I can't remember too well though I think this bug is coming from group validations, or logic that was introduced to handle group validations. In the diagram above, the pink circles are Group validations which are dependant on each other, so in that case, the execution flow must wait until all group members (input fields) are ready to execute - hence the "Field is pending..." message.

**Comment by jpillora** [comment]
https://github.com/jpillora/verifyjs/wiki/Contributor-Documentation

**Comment by CAYdenberg** [comment]
@ht512512 could you post the URL for the script you have been using? I tried the version (of both notify.js and verify.js) that is included on the Verify.js website (http://verifyjs.com/js/notify.js, http://verifyjs.com/js/verify.js) and didn't have any luck.
Casey

**Comment by ht512512** [comment]
@CAYdenberg 
Sorry, I don't have that handy anymore since I moved away from that workaround and went back to the latest version. The better work around for me was to use ajax submit. That works fine with the latest version of verify-notify.js. You do have to call verify() method from javascript though.

**Comment by CAYdenberg** [comment]
Yes, that's what I've ended up doing, too. Would be nice to have the fallback but I guess if JS is disabled verify just won't work :)

**Comment by ht512512** [comment]
@CAYdenberg 
Agree. 
Just noticed an error in my workaround description. I meant to say, "You do have to call validate() method from javascript though."

**Comment by ghost** [comment]
I have a same issue, anyone with luck?

![2014-12-01 11 08 29](https://cloud.githubusercontent.com/assets/9945832/5240523/6e0de144-794a-11e4-8197-6362e4c00821.png)

**Comment by Inventame** [comment]
Hi, i had a same issue, i could not fixed up, but i been able to avoid it, reviewing my code i noted a button inside the form, when i remove this button everything works. I know it is not a solution but works for me. Maybe you have buttons or other object inside the form who are making it happen

**Comment by Inventame** [comment]
Hi, the library works like a charm, my mistake was use the same id in two objects inside the form, when this mistake happen this issue appear.

**Comment by Inventame** [comment]
I found a solution, in the object "ValidationForm" on the "init" method remove the line "$(document).ready(this.domReady);" and it will work. Was the solution for me

**Comment by raul1991** [comment]
No it wont work i tried.,,,

**Comment by vladzaets** [comment]
I have the same issue :-(

**Comment by raul1991** [comment]
This usually happens when you have hidden fields in your form. When you take them outside the form it works(atleast in my case). @vladzaets

**Comment by zuzuleinen** [comment]
I fixed by commenting out the following code:

```javascript
      if(!ruleParams.required && !$.trim(this.domElem.val())) {
        this.warn("skip (not required)");
        return true;
      }
```
